source/Irrlicht/COpenGLFunctionTable.h

```diff
@@ -0,0 +1,4530 @@
+		{
+			if (pGlEndQuery)
+				pGlEndQuery(target);
+		}
+
+		inline void COpenGLFunctionTable::extGlBeginQueryIndexed(GLenum target, GLuint index, GLuint id)
+		{
+			if (pGlBeginQueryIndexed)
+				pGlBeginQueryIndexed(target, index, id);
+		}
+
+		inline void COpenGLFunctionTable::extGlEndQueryIndexed(GLenum target, GLuint index)
+		{
+			if (pGlEndQueryIndexed)
+				pGlEndQueryIndexed(target, index);
+		}
+
+
+		inline void COpenGLFunctionTable::extGlGetQueryObjectuiv(GLuint id, GLenum pname, GLuint* params)
+		{
+			if (pGlGetQueryObjectuiv)
+				pGlGetQueryObjectuiv(id, pname, params);
+		}
+
+		inline void COpenGLFunctionTable::extGlGetQueryObjectui64v(GLuint id, GLenum pname, GLuint64* params)
+		{
+			if (pGlGetQueryObjectui64v)
+				pGlGetQueryObjectui64v(id, pname, params);
+		}
+
+		inline void COpenGLFunctionTable::extGlGetQueryBufferObjectuiv(GLuint id, GLuint buffer, GLenum pname, GLintptr offset)
+		{
+			if (Version < 440 && !FeatureAvailable[IRR_ARB_query_buffer_object])
+			{
+#ifdef _DEBuG
+				os::Printer::log("GL_ARB_query_buffer_object unsupported!\n");
+#endif // _DEBuG
+				return;
+			}
+
+			if (Version >= 450 || FeatureAvailable[IRR_ARB_direct_state_access])
+			{
+				if (pGlGetQueryBufferObjectuiv)
+					pGlGetQueryBufferObjectuiv(id, buffer, pname, offset);
+			}
+			else
+			{
+				GLint restoreQueryBuffer;
+				glGetIntegerv(GL_QUERY_BUFFER_BINDING, &restoreQueryBuffer);
+				pGlBindBuffer(GL_QUERY_BUFFER, id);
+				if (pGlGetQueryObjectuiv)
+					pGlGetQueryObjectuiv(id, pname, reinterpret_cast<GLuint*>(offset));
+				pGlBindBuffer(GL_QUERY_BUFFER, restoreQueryBuffer);
+			}
+		}
+
+		inline void COpenGLFunctionTable::extGlGetQueryBufferObjectui64v(GLuint id, GLuint buffer, GLenum pname, GLintptr offset)
+		{
+			if (Version < 440 && !FeatureAvailable[IRR_ARB_query_buffer_object])
+			{
+#ifdef _DEBuG
+				os::Printer::log("GL_ARB_query_buffer_object unsupported!\n");
+#endif // _DEBuG
+				return;
+			}
+
+			if (Version >= 450 || FeatureAvailable[IRR_ARB_direct_state_access])
+			{
+				if (pGlGetQueryBufferObjectui64v)
+					pGlGetQueryBufferObjectui64v(id, buffer, pname, offset);
+			}
+			else
+			{
+				GLint restoreQueryBuffer;
+				glGetIntegerv(GL_QUERY_BUFFER_BINDING, &restoreQueryBuffer);
+				pGlBindBuffer(GL_QUERY_BUFFER, id);
+				if (pGlGetQueryObjectui64v)
+					pGlGetQueryObjectui64v(id, pname, reinterpret_cast<GLuint64*>(offset));
+				pGlBindBuffer(GL_QUERY_BUFFER, restoreQueryBuffer);
+			}
+		}
+
+		inline void COpenGLFunctionTable::extGlQueryCounter(GLuint id, GLenum target)
+		{
+			if (pGlQueryCounter)
+				pGlQueryCounter(id, target);
+		}
+
+		inline void COpenGLFunctionTable::extGlBeginConditionalRender(GLuint id, GLenum mode)
+		{
+			if (pGlBeginConditionalRender)
+				pGlBeginConditionalRender(id, mode);
+		}
+
+		inline void COpenGLFunctionTable::extGlEndConditionalRender()
+		{
+			if (pGlEndConditionalRender)
+				pGlEndConditionalRender();
+		}
```

Comment:
all the functions that don't have special workarounds, i.e. check for `Version` to choose between different function calls to use, etc.

You can remove these functions and make the variables like `pGlEndConditionalRender` public, BEWARE, for stuff that needs the fallbacks and special code... DO NOT MAKE THE FUNCTION POINTERS PUBLIC!